usaspending_api/broker/management/commands/load_fpds_transactions.py

```diff
@@ -166,6 +166,13 @@ def handle(self, *args, **options):
             )
             update_c_to_d_linkages("contract")
 
+        if failed_ids:
```

Comment:
Good idea, we probably want the log messages to be `error` instead of `info` though. Also the script could return a non-zero exit code. using `raise SystemExit(<non-zero integer>)`

Reply:
If we had the program exit on failure, do we want to still update the last_load_date? I would think not since that would leave the failed awards lost forever, but not updating it risks the incremental loader falling far out of date if the warnings aren't noticed over a long weekend.

Reply:
Just spoke to ops. Adding a non-zero code will require updates to groovy scripts, so we might want to leave this enhancement as future work with its own ticket.

Reply:
> If we had the program exit on failure, do we want to still update the last_load_date? I would think not since that would leave the failed awards lost forever
I agree with not updating the last load date if errors are encountered.

> but not updating it risks the incremental loader falling far out of date if the warnings aren't noticed over a long weekend.
That's where the non-zero exit code comes in, the pipeline will stop on the FPDS stage and report a failure. Makes it easier for night watch

> Just spoke to ops. Adding a non-zero code will require updates to groovy scripts, so we might want to leave this enhancement as future work with its own ticket.
OPS is talking about are when there is a non-fatal exit code which should allow the pipeline to continue. I am proposing an exit code which would stop the entire pipeline. No updates to groovy scripts are necessary since the default behavior is the desired behavior.

Reply:
Is stopping the entire pipeline a good idea? That will add ops backlog for all the other tasks that wouldn't run that night, including not refreshing any matviews. If that's really what we want, we could just not add any of this functionality, and let the loader crash on the line with the problem. If we have to rerun the whole load any time there's an issue, we don't get much by carrying on in this stage and listing issues afterwards.

Reply:
> Is stopping the entire pipeline a good idea

Yes when a critical stage is unsuccessful. So much of the latter stages (and the website as a whole) are based on transaction data. If there are failures, the pipeline should die so the issues can be addressed immediately. If the pipeline reports an overall success, who is going to diligently look at the logs after each run for failures?

> That will add ops backlog for all the other tasks that wouldn't run that night, including not refreshing any matviews

It isn't adding anything, since the current loader will stop the rest of the pipeline. This falls under the night-watch responsibilities, not operations.

> If that's really what we want, we could just not add any of this functionality, and let the loader crash on the line with the problem

Up to you

> If we have to rerun the whole load any time there's an issue, we don't get much by carrying on in this stage and listing issues afterwards.

Logging the failed IDs and allowing the successful ones to be stored in the DB provides options. Currently the only remedy is to re-run the script across all of the original records since the entire transaction is rolled back. The changes were are discussing could allow someone to attempt to reload only the failed transactions to "get-back-to-good" and if the specific load is successful the last-load-date can be manually updated and the pipeline can be restarted on the next stage.